### PR TITLE
Feat/remove individual filters

### PR DESCRIPTION
**Pull request by BLuEScioN**
## What type of PR is this? (check all applicable)
- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Adds UI onto the filters to remove them individually

## Issue ticket number and link
closes  https://github.com/hirosystems/explorer/issues/2218

# Checklist:
- [X] I have performed a self-review of my code.
- [X] I have tested the change on desktop and mobile.
- [ ] I have added thorough tests where applicable.
- [ ] I've added analytics and error logging where applicable.

### CODE REVIEW

**Comment by ginny-d** [comment]
Thanks @BLuEScioN, it looks & works great!

Just one small request if possible:
Changing the X icon color from `icon-tertiary` to `icon-secondary`.

Reasoning:
While Figma specifies icon-tertiary, the implementation uses the regular icon weight instead of bold (used in Figma). This results in reduced contrast and visibility. Updating to `icon-secondary` should help align the visual clarity with the original design intent.

File: src/components/ui/popover.tsx [hunk 1]
```diff
@@ -49,6 +49,13 @@ export const PopoverCloseTrigger = React.forwardRef<
   );
 });
 
+export const PopoverTriggerWithClear = React.forwardRef<
+  HTMLButtonElement,
+  ChakraPopover.TriggerProps
+>(function PopoverTriggerWithClear(props, ref) {
+  return <ChakraPopover.Trigger {...props} ref={ref} />;
+});
+
```
Comment (He1DAr) [inline src/components/ui/popover.tsx]: would it work with something like:
```suggestion
export const PopoverTrigger = ChakraPopover.Trigger;
```

File: src/common/components/table/filters/table-filters-utils.tsx [hunk 2]
```diff
@@ -60,6 +60,25 @@ export const useDateFilterHandler = (useShallow: boolean = true) => {
...
+  const params = new URLSearchParams(searchParams);
+  const searchParamsWithoutDateFilter = removeDateFilterFromSearchParams(params);
+  if (useShallow) {
+    return () => shallowRouter.replace(null, '', `?${searchParamsWithoutDateFilter.toString()}`);
+  } else {
+    return () => router.replace(`?${searchParamsWithoutDateFilter.toString()}`);
+  }
+};
+
 export const getAddressFilterParams = (
```
Comment (He1DAr) [inline src/common/components/table/filters/table-filters-utils.tsx]: I'm a bit unclear on the intent of this function, name suggests it's a getter but it can set or delete a params.

File: src/common/components/table/filters/table-filters-utils.tsx [hunk 3]
```diff
@@ -60,6 +60,25 @@ export const useDateFilterHandler = (useShallow: boolean = true) => {
   };
```
Comment (He1DAr) [inline src/common/components/table/filters/table-filters-utils.tsx]: This file seems to have a repeated pattern across all filter handlers, each one follows similar structure: building URLSearchParams, applying changes, and updating the URL (router.replace or shallowRouter.replace)

Would you consider abstracting this into a generic testable utility (e.g. useQueryUpdater) that takes a param mutation function and handles the routing logic? It might help reduce duplication and make it easier to add new filters in the future. Also maybe putting all filter keys/query param into enum for better typing.

File: src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterAccordionItem.tsx [hunk 4]
```diff
@@ -1,48 +1,75 @@
...
+  //       p={1.5} // I think there is a bug on Chakra that's causing the padding here to be applied to 2 divs surrounding the content
+  //     >
+  //       <TransactionTypeFilterForm
+  //         open={open}
+  //         onSubmit={onSubmit}
+  //         defaultTransactionType={transactionType}
+  //       />
+  //     </AccordionItemContent>
+  //   </AccordionItem>
+  // );
```
Comment (He1DAr) [inline src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterAccordionItem.tsx]:
```suggestion
```

File: src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterPopover.tsx [hunk 5]
```diff
@@ -1,30 +1,37 @@
...
         placement: 'bottom-start',
         offset: { mainAxis: TAB_HEIGHT_ADJUSTMENT },
         sameWidth: true,
       }}
       trigger={(open, setOpen) => (
-        <TransactionTypeFilterTriggerText open={open} transactionType={defaultTransactionType} />
+        <TransactionTypeFilterTrigger
+          open={open}
+          transactionType={defaultTransactionType}
+          clearFilterHandler={clearFilterHandler}
```
Comment (He1DAr) [inline src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterPopover.tsx]: Can `useTxTableFilters` be used in component directly to avoid prop drilling?

File: src/common/components/table/filters/table-filters-utils.tsx [hunk 6]
```diff
@@ -94,6 +119,21 @@ export function useAddressFilterHandler(useShallow: boolean = true) {
...
 }
 
+export const useClearAddressFilterHandler = (useShallow: boolean = true) => {
```
Comment (He1DAr) [inline src/common/components/table/filters/table-filters-utils.tsx]: It looks like all clear hooks follow same pattern, we could maybe have 1 hook:
```
export const useClearFilterHandler = (removeFn, useShallow) => {
  const searchParams = useSearchParams();
  const shallowRouter = useShallowRouter();
  const router = useRouter();

  return () => {
    const params = new URLSearchParams(searchParams);
    const updatedParams = removeFn(params);
    const query = `?${updatedParams.toString()}`;
    useShallow
      ? shallowRouter.replace(null, '', query)
      : router.replace(query);
  };
};
```
then we reuse it like:
```
export const useClearDateFilterHandler = (useShallow) =>
  useClearFilterHandler(removeDateFilterFromSearchParams, useShallow);

export const useClearAddressFilterHandler = (useShallow) =>
  useClearFilterHandler(removeAddressFilterFromSearchParams, useShallow);
  
....
```

File: src/common/components/table/filters/table-filters-utils.tsx (hunk 3 above)
Comment (He1DAr) [inline src/common/components/table/filters/table-filters-utils.tsx]: I think adding a unit test suite for this file would be really valuable to help catch edge cases and give us confidence updating it in future.

**Review by He1DAr — CHANGES_REQUESTED** [review]
Great work! Left a few comments around potential abstractions and testing.

File: src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterAccordionItem.tsx (hunk 4 above)
Comment (BLuEScioN) [reply src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterAccordionItem.tsx]: done

File: src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterPopover.tsx (hunk 5 above)
Comment (BLuEScioN) [reply src/common/components/table/filters/transaction-type-filter/TransactionTypeFilterPopover.tsx]: The reason I am not doing that is because I want these filter components to be more abstracted so that they are reusable across the explorer, on other pages, such as the transaction id page, or the blocks page. If I use the useTxTableFiters context hook in them then I am tying their implementation to the TxTable.

File: src/components/ui/popover.tsx (hunk 1 above)
Comment (BLuEScioN) [reply src/components/ui/popover.tsx]: This isn't used anywhere. I think I was just playing around with an idea

File: src/common/components/table/filters/table-filters-utils.tsx (hunk 2 above)
Comment (BLuEScioN) [reply src/common/components/table/filters/table-filters-utils.tsx]: I see your point. Those functions have been removed, and I have abstracted the query manipulation functionality into a hook called useQueryUpdater, and I have simplified the logic for handling query params for each filter into individual filter mutator functions.

File: src/common/components/table/filters/table-filters-utils.tsx (hunk 3 above)
Comment (BLuEScioN) [reply src/common/components/table/filters/table-filters-utils.tsx]: done

Comment (BLuEScioN) [reply src/common/components/table/filters/table-filters-utils.tsx]: done
